redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,692 @@
+        asyncResp->res.jsonValue["@odata.id"] = "/redfish/v1/LicenseService/";
+        asyncResp->res.jsonValue["@odata.type"] =
+            "#LicenseService.v1_0_0.LicenseService";
+        asyncResp->res.jsonValue["Name"] = "License Service";
+        asyncResp->res.jsonValue["Id"] = "LicenseService";
+
+        asyncResp->res.jsonValue["Licenses"] = {
+            {"@odata.id", "/redfish/v1/LicenseService/Licenses"}};
+        asyncResp->res.jsonValue["Actions"] = {
+            {"#LicenseService.Install",
+             {{
+                 "target",
+                 "/redfish/v1/LicenseService/Actions/"
+                 "LicenseService.Install",
+             }}}};
+    });
+}
+
+static void resetLicenseActivationStatus(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    std::string value{"com.ibm.License.LicenseManager.Status.Pending"};
+    crow::connections::systemBus->async_method_call(
+        [asyncResp](const boost::system::error_code ec) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error: Unable to set "
+                             "the LicenseString property {}",
+                             ec);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        getLicenseActivationStatusMatch() = nullptr;
+    },
+        "com.ibm.License.Manager", "/com/ibm/license",
+        "org.freedesktop.DBus.Properties", "Set",
+        "com.ibm.License.LicenseManager", "LicenseActivationStatus",
+        std::variant<std::string>(value));
+}
+
+static void
+    resetLicenseString(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    std::string value;
+    crow::connections::systemBus->async_method_call(
+        [asyncResp](const boost::system::error_code ec) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error: Unable to set "
+                             "the LicenseString property {}",
+                             ec);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        resetLicenseActivationStatus(asyncResp);
+    },
+        "com.ibm.License.Manager", "/com/ibm/license",
+        "org.freedesktop.DBus.Properties", "Set",
+        "com.ibm.License.LicenseManager", "LicenseString",
+        std::variant<std::string>(value));
+}
+
+inline void
+    getLicenseActivationAck(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                            const std::string& status,
+                            const std::string& licenseString)
+{
+    if (status == "com.ibm.License.LicenseManager.Status.ActivationFailed")
+    {
+        BMCWEB_LOG_ERROR("LicenseActivationStatus: ActivationFailed");
+        messages::installFailed(asyncResp->res, "ActivationFailed");
+    }
+    else if (status == "com.ibm.License.LicenseManager.Status.InvalidLicense")
+    {
+        BMCWEB_LOG_ERROR("LicenseActivationStatus: InvalidLicense");
```

Comment:
This is a user error right? This should be a warning..

Reply:
moved to warning